src/repo.rs

```diff
@@ -161,6 +161,74 @@ impl Repo {
         self.update_hash_on_dht(&hash).await?;
         Ok(hash)
     }
+
+    // Method to get or create a collection associated with the repo
+    pub async fn get_or_create_collection(&self) -> Result<Hash> {
+        self.check_write_permissions()?;
+        let collection_name = self.get_name().await?;
+
+        match self.iroh_blobs.collection_hash(&collection_name).await {
+            Ok(collection_hash) => Ok(collection_hash),
+            Err(_) => self.iroh_blobs.create_collection(&collection_name).await,
+        }
+    }
+
+    // Method to add a file to the collection identified by repo ID
+    pub async fn set_file_in_repo_collection(&self, file_name: &str, file_hash: Hash) -> Result<Hash> {
+        self.check_write_permissions()?;
+        let collection_name = self.get_name().await?;
+
+        self.iroh_blobs.set_file(&collection_name, file_name, &file_hash).await
+    }
+
+    // Method to retrieve a file's hash from the collection
+    pub async fn get_file_from_repo_collection(&self, file_name: &str) -> Result<Hash> {
+        let collection_name = self.get_name().await?;
+
+        self.iroh_blobs.get_file(&collection_name, file_name).await
+    }
+
+    // Method to list all files in the collection
+    pub async fn list_files_in_repo_collection(&self) -> Result<Vec<String>> {
```

Comment:
```suggestion
    async fn list_files_in_repo_collection(&self) -> Result<Vec<String>> {
```

This should be called in the top level `list_files` if can_Write == true and kept private as an implementation detail

Reply:
e52e9bc672d5cdd8dffd43b784ee3928afdf170b